modules/imgcodecs/src/loadsave.cpp

```diff
@@ -421,14 +421,14 @@ imread_( const String& filename, int flags, int hdrtype, Mat* mat=0 )
     }
 
     int scale_denom = 1;
-    if( flags > IMREAD_LOAD_GDAL )
+    if( ( flags > IMREAD_LOAD_GDAL ) & ( flags < IMREAD_IGNORE_ORIENTATION ) )
```

Comment:
if( flags > IMREAD_LOAD_GDAL ) is good here ( to make usage like `IMREAD_REDUCED_GRAYSCALE_4 | IMREAD_IGNORE_ORIENTATION`  acceptable )
I recommend that you do not make any changes to the `imread ` function

Reply:
Sure @sturkmen72, I'll remove it.